### PR TITLE
🐢Assert Android advanced Scheduling?

### PR DESCRIPTION
**Pull request by timrijckaert**
It is somewhat unclear that you can call `Workmanager.registerPeriodicTask` and `Workmanager.registerOneOffTask` on `iOS` eventhought this does not do anything.  

Open for debate: Should the library throw an Error when you try to call Android specific calls, or is the current implementation OK?

### CODE REVIEW

File: lib/workmanager.dart [hunk 1]
```diff
@@ -235,15 +236,28 @@ class Workmanager {
...
   /// Cancels all tasks
-  static Future<void> cancelAll() async =>
-      await _foregroundChannel.invokeMethod("cancelAll");
+  static Future<void> cancelAll() async {
+    _assertNotIOS();
+    await _foregroundChannel.invokeMethod("cancelAll");
+  }
+
+  static void _assertNotIOS() {
+    if (Platform.isIOS) {
```
Comment (petermnt) [inline lib/workmanager.dart]: How about assert(Platform.isIOS) ?

Comment (petermnt) [reply lib/workmanager.dart]: To clarify, I was thinking of this for 2 reasons.
* Asserting that we are on Android makes more sense.
* Assert only evaluates/throws in debug

Comment (timrijckaert) [reply lib/workmanager.dart]: Yes, I think I made a boo boo there it should either be `!Platform.isIOS` or `Platform.isAndroid`.
Question remains, is an AssertionError enough in debug or do we want to throw an Exception?

Comment (petermnt) [reply lib/workmanager.dart]: AssertionError sounds good to me. It's commonly used in the Flutter framework for 'illegal states'.

**Comment by timrijckaert** [comment]
Closing for now since we will probably need to re-investigate how we will build our public Dart API

**Pull request closed without merging**